extension.ts

```diff
@@ -127,6 +127,11 @@ export async function activate(context: vscode.ExtensionContext) {
     }, 0);
   });
 
+  vscode.workspace.onDidCloseTextDocument((event) => {
+    // Remove modehandler for closed document
+    delete modeHandlerToEditorIdentity[event.fileName + vscode.window.activeTextEditor.viewColumn];
```

Comment:
It is created in getAndUpdateModehandler based on ID.

If a more handler with that I'd already exists it will use it and if not it creates a new one. You can see this by creating a new file, entering insert, then closing the file and creating a new untitled file(must have same name ex. Untitled-1.)

This just removes a MH after a tab is closed.

This mitigates if bad bugs happen, you have a chance of closing and opening the tab instead of restarting vscode.